### DISCUSSION TITLE
Email is sent but rejected by outlook because of the container's ipv6

### DISCUSSION THREAD

**Discussion by MrChausson** [dani-garcia/vaultwarden#4342] [Q&A] [answered]
### Subject of the issue

Hello, so I have just installed my vaultwarden (with the latest docker image) on my k3s cluster. And I have managed to make SMTP to work with office365 but the test mails are coming back to my inbox by outlook with this error:
```
Generating server: MR1P264MB3282.FRAP264.PROD.OUTLOOK.COM

lol.lool2546@yopmail.com
Remote server returned '550 5.7.708 Service unavailable. Access denied, traffic not accepted from this IP. For more information please go to http://go.microsoft.com/fwlink/?LinkId=526653 AS(7230) [PR0P264MB3387.FRAP264.PROD.OUTLOOK.COM 2024-02-12T13:14:51.999Z 08DC2A79F18677C8]'

Original message headers:

Received: from MR1P264MB3282.FRAP264.PROD.OUTLOOK.COM
 ([fe80::7811:d5d9:5666:93ee]) by MR1P264MB3282.FRAP264.PROD.OUTLOOK.COM
 ([fe80::7811:d5d9:5666:93ee%4]) with mapi id 15.20.7270.036; Mon, 12 Feb 2024
 13:14:52 +0000
MIME-Version: 1.0
Content-Type: text/plain
Date: Mon, 12 Feb 2024 13:14:52 +0000
Message-ID: <096578bd-e605-4707-9e6a-32b9fa1c2a78@chausson.services>
Subject: Vaultwarden SMTP Test
```

The test emails are appearing in my Junk Email section of outlook:
![image](https://github.com/dani-garcia/vaultwarden/assets/28267018/dd9fbcde-57f5-4096-924e-d0cfaa3ae5ff)

It looks like it is using a temporary local ipv6 address on the internet (which has bad reputation because it has already been used a lot).

Is it possible to modify the image in order to not use this local address on the internet ?

### Deployment environment

* vaultwarden server docker version: vaultwarden/server:1.30.3

* Install method: with this helm chart: https://artifacthub.io/packages/helm/gissilabs/vaultwarden on kubernetes v1.26.10+k3s2

* Not that the cluster I have installed it on is using dual-stack networking which means containers have ipv4 and ipv6 internet access.

* Reverse proxy and version: traefik

### Steps to reproduce

- Install this helm chart: https://artifacthub.io/packages/helm/gissilabs/vaultwarden
- Change the smtp settings to make it work with office365
- Change the tag to use the version: vaultwarden/server:1.30.3

### Expected behaviour

- Connect as admin
- Send a test email
- Email is sent and not rejected by outlook
- Recipient receives the mail

### Actual behaviour

- Connect as admin
- Send a test email
- Email is sent but rejected by outlook which send it back to my mailbox with the message shown above
- Recipient never receives the mail
### Troubleshooting data

The ipv6 rejected by outlook looks to be a local ipv6 this is why I think it is rejected.

Thanks for your help in advance

**Comment by BlackDex** [comment]
I actually think it has nothing to do with Vaultwarden, but more how the helm-chart/k8s works in this way.
Vaultwarden just uses the interface it is provided, and if that interface has a temp IPv6 address that is not something we can fix.

Same goes for IPv4 it just uses the provided IP's of the interface dedicated to Vaultwarden (or which every other Pod you run).

**Comment by MrChausson** [comment]
Okay I see, I will search how to change that interface ip address then.

Thanks

**Comment by MrChausson** [comment, marked as answer]
In case someone comes across this; The problem was I was using a free tenant trial for developers of office365 ([this thing](https://developer.microsoft.com/en-us/microsoft-365/dev-program)) and there is a limit of how many mails you can send using SMTP, so I think my account was flagged because I used it too much.

I did not setup it for vaulwarden yet but to fix this problem for my others projects using SMTP I used a [free mailjet account](https://www.mailjet.com/pricing/) which worked well for me.
I think it could work on Vaultwarden using the [Mailjet SMTP RELAY](https://dev.mailjet.com/smtp-relay/overview/)
